core/src/main/java/com/turn/ttorrent/common/ByteBufferPool.java

```diff
@@ -0,0 +1,55 @@
+package com.turn.ttorrent.common;
+
+import org.apache.commons.pool2.BasePooledObjectFactory;
+import org.apache.commons.pool2.PooledObject;
+import org.apache.commons.pool2.impl.DefaultPooledObject;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
+
+import java.nio.ByteBuffer;
+
+public class ByteBufferPool {
+
+    private final GenericObjectPool<ByteBuffer> pool;
+
+    public ByteBuffer borrowObject() throws Exception {
```

Comment:
Since it doesn't look like you're doing much with that checked Exception is the upper layers anyway, it might be simpler to catch it here and propagate it as a RuntimeException instead (similar to what `Throwables.propagate()` would do.